client/src/run.rs

```diff
@@ -314,23 +257,11 @@ impl Run {
                 key: k.clone(),
                 ..Default::default()
             };
-            match v {
-                Value::Ndarray(arr) => {
-                    // TODO: convert to image if shape is valid, otherwise just serialize
-                    let shape = arr.shape();
-                    if shape.len() == 3 {
-                        let value_json = ndarray_to_image(arr, &self.settings.files_dir());
-                        item.value_json = serde_json::to_string(&value_json).unwrap();
-                        // TODO: tell nexus to upload the image
-                        self.save_files(&value_json.get("path").unwrap().to_string());
-                    } else {
-                        item.value_json = serde_json::to_string(&Value::Ndarray(arr)).unwrap();
-                    }
-                }
-                _ => {
-                    item.value_json = serde_json::to_string(&v).unwrap();
-                }
+            // TODO: this needs to be recursive...
+            if v.is_object() && v.get("_type").unwrap() == "image-file" {
```

Comment:
Here's the custom type detection in the core library.  Doing it generically like this would also enable implementing serialization directly in the library while still supporting file syncing etc. in rust land.  Note, this is only handling top level values, we'll need to make this recurse through nested dicts.